pkg/db/db.go

```diff
@@ -0,0 +1,413 @@
+	err := row.Scan(&tbl, &createTable)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	if !tbl.Valid || !createTable.Valid {
+		return "", errors.NewNotFound(nil, fmt.Sprintf("table %s not exist", tableName))
+	}
+	return createTable.String, nil
+}
+
+// GetCount get count rows of the table for specific field.
+func GetCount(db *sql.DB, dbname string, table string, where string) (int64, error) {
+	query := fmt.Sprintf("SELECT count(1) cnt from `%s`.`%s` where %s", dbname, table, where)
+	rows, err := db.Query(query)
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+	defer rows.Close()
+
+	var fields map[string][]byte
+	if rows.Next() {
+		fields, err = ScanRow(rows)
+	}
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+	cntStr, ok := fields["cnt"]
+	if !ok {
+		return 0, errors.New("[dumper] `cnt` field not found in select count sql result")
+	}
+	cnt, err := strconv.ParseInt(string(cntStr), 10, 64)
+	return cnt, errors.Trace(err)
+}
+
+// ShowIndex returns result of execute `show index`
+func ShowIndex(db *sql.DB, dbname string, table string) ([]map[string][]byte, error) {
+	/*
+		show index example result:
+		mysql> show index from test;
+		+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
+		| Table | Non_unique | Key_name | Seq_in_index | Column_name | Collation | Cardinality | Sub_part | Packed | Null | Index_type | Comment | Index_comment |
+		+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
+		| test  | 0          | PRIMARY  | 1            | id          | A         | 0           | NULL     | NULL   |      | BTREE      |         |               |
+		| test  | 0          | aid      | 1            | aid         | A         | 0           | NULL     | NULL   | YES  | BTREE      |         |               |
+		+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
+	*/
+
+	query := fmt.Sprintf("SHOW INDEX FROM `%s`.`%s`", dbname, table)
+	rows, err := db.Query(query)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	defer rows.Close()
+
+	var rowsData []map[string][]byte
+	for rows.Next() {
+		fields, err1 := ScanRow(rows)
+		if err1 != nil {
+			return nil, errors.Trace(err1)
+		}
+		rowsData = append(rowsData, fields)
+	}
+
+	return rowsData, nil
+}
+
+func FindSuitableIndex(db *sql.DB, dbName string, table string) (*model.ColumnInfo, error) {
+	rowsData, err := ShowIndex(db, dbName, table)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	tableInfo, err := GetSchemaTable(db, dbName, table)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	// seek pk
+	for _, fields := range rowsData {
+		if string(fields["Key_name"]) == "PRIMARY" && string(fields["Seq_in_index"]) == "1" {
```

Comment:
do u want to find first column in the index?

Reply:
`show index` statement supports `where` expression.
https://dev.mysql.com/doc/refman/5.7/en/extended-show.html

e.g.

show index from `xxx` WHERE Key_name='PRIMARY';
show index from `xxx` WHERE Non_unique='0';